### PR TITLE
fix: Tags Page Polish

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY
Updating tags page with request from product. The following changes were made:
- empty state if no entities are used by a tag
- fix loading flicker
- added new columns
- new styles from spec

#### After
Screen Shot 2023-09-29 at 3 32 32 PM
 design decisions -->

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/allEntities/AllEntitiesTable.tsx [hunk 1]
```diff
@@ -86,10 +112,14 @@ export default function AllEntitiesTable({
   }, [search]);
 
   const renderTable = (type: objectType) => {
+    console.log(objects);
```
Comment (betodealmeida) [inline superset-frontend/src/features/allEntities/AllEntitiesTable.tsx:115]:
```suggestion
```

File: superset-frontend/src/features/allEntities/AllEntitiesTable.tsx [hunk 2]
```diff
@@ -99,27 +129,71 @@ export default function AllEntitiesTable({
...
-        <h3>{t('Charts')}</h3>
-        {renderTable('chart')}
-        <hr />
-        <h3>{t('Queries')}</h3>
-        {renderTable('query')}
-      </AllEntitiesTableContainer>
-    );
-  }
-  return <Loading />;
+  console.log(objects);
```
Comment (betodealmeida) [inline superset-frontend/src/features/allEntities/AllEntitiesTable.tsx:175]:
```suggestion
```

**Review by betodealmeida — COMMENTED** [review]
@hughhhh can you add some screenshots, so people outside of Preset can know what changes this PR is introducing?

**Comment by justinpark** [comment]
@hughhhh Thanks for updates. Could you add the screenshot for the update in the description?

File: superset-frontend/src/features/allEntities/AllEntitiesTable.tsx [hunk 3]
```diff
@@ -99,27 +131,69 @@ export default function AllEntitiesTable({
...
+                };
+              };
+            }) => (
+              // Only show custom type tags
+              <TagsList
+                tags={tags.filter(
+                  (tag: Tag) =>
+                    tag.type === 'TagTypes.custom' || tag.type === 1,
+                )}
+                maxTags={3}
```
Comment (lilykuang) [inline superset-frontend/src/features/allEntities/AllEntitiesTable.tsx:154]: Can this number be a constant on top of the file?

File: superset-frontend/src/features/allEntities/AllEntitiesTable.tsx [hunk 4]
```diff
@@ -22,18 +22,31 @@ import { t, styled, logging } from '@superset-ui/core';
...
   .table {
     table-layout: fixed;
   }
   .td {
     width: 33%;
   }
+  .entity-title {
+    font-family: Inter;
+    font-size: 14px;
+    font-weight: 500;
```
Comment (lilykuang) [inline superset-frontend/src/features/allEntities/AllEntitiesTable.tsx:44]:
```suggestion
    font-size: ${({ theme }) => theme.typography.sizes.m}px;
    font-weight: ${({ theme }) => theme.typography.weights.medium};
```

**Review by lilykuang — APPROVED** [review]
small nits. LGMT!

**Pull request merged**